Code/Engine/Core/Graphics/Geometry.h

```diff
@@ -145,29 +164,26 @@ class EZ_CORE_DLL ezGeometry
 
   /// \brief Adds a rectangle shape in the XY plane, with the front in positive Z direction.
   /// It is centered at the origin, extending half size.x and half size.y into direction +X, -X, +Y and -Y.
-  void AddRectXY(const ezVec2& size, const ezColor& color, const ezMat4& mTransform = ezMat4::IdentityMatrix(), ezUInt16 boneIndex = 0);
-
-  /// \brief Same as AddRectXY but additionally tessellates the plane. Tessellation factors must be larger than zero.
-  void AddTesselatedRectXY(const ezVec2& size, const ezColor& color, ezUInt32 uiTesselationX, ezUInt32 uiTesselationY, const ezMat4& mTransform = ezMat4::IdentityMatrix(), ezUInt16 boneIndex = 0);
+  /// Optionally tessellates the rectangle for more detail.
+  void AddRectXY(const ezVec2& size, ezUInt32 uiTesselationX = 1, ezUInt32 uiTesselationY = 1, const GeoOptions& options = GeoOptions());
 
-  /// \brief Adds an untextured box (8 vertices).
-  void AddBox(const ezVec3& size, const ezColor& color, const ezMat4& mTransform = ezMat4::IdentityMatrix(), ezUInt16 boneIndex = 0);
+  /// \brief Adds a box.
+  /// If bExtraVerticesForTexturing is false, 8 shared vertices are added.
+  /// If bExtraVerticesForTexturing is true, 24 separate vertices with UV coordinates are added.
+  void AddBox(const ezVec3& size, bool bExtraVerticesForTexturing, const GeoOptions& options = GeoOptions());
```

Comment:
I found AddBox and AddTexturedBox easier to understand than AddBox(..., false) and AddBox(..., true)

Reply:
Yeah, interfaces are hard. I've merged them together, because this way it is easier to discover that there even are different options. I might do the same with cylinder/solid-cylinder.